File: datastruct/map/Map.java
public interface Map<K,V> {

    // 向映射集中添加新的键值对
    void add(K key, V value);

    // 删除键为 key 的键值对并返回 value
    V remove(K key);

    // 是否包含键为 key 的键值对
    boolean contain(K key);

    // 将键 key 的值设为 value
    void set(K key, V value);

    // 返回键 key 对应的值 value
    V get(K key);

    // 映射集大小
    int getSize();

    // 映射集是否为空
    boolean isEmpty();

}

File: datastruct/map/Map.py
from abc import ABCMeta, abstractmethod


class Map(metaclass=ABCMeta):

    @abstractmethod
    def contains(self, key):
        """映射中是否包含 key"""
        pass

    @abstractmethod
    def is_empty(self):
        """映射是否为空"""
        pass

    @abstractmethod
    def get_size(self):
        """映射大小"""
        pass

    @abstractmethod
    def add(self, key, value):
        """向映射添加新（key，value）对，如果 key 已存在，更新 value"""
        pass

    @abstractmethod
    def remove(self, key):
        """从映射中删除 key 所在的结点"""
        pass

    @abstractmethod
    def get(self, key):
        """获取 key 所对应的 value，否则返回 None"""
        pass

    @abstractmethod
    def set(self, key, value):
        """设置 key 对应值为 value"""
        pass
